iothub/device/src/DirectMethod/EdgeModuleDirectMethodRequest.cs

```diff
@@ -23,25 +26,76 @@ public EdgeModuleDirectMethodRequest(string methodName)
         /// a direct method on an edge device or an edge module connected to the same edge hub.
         /// </summary>
         /// <param name="methodName">The method name to invoke.</param>
-        /// <param name="payload">The direct method payload bytes.</param>
-        public EdgeModuleDirectMethodRequest(string methodName, byte[] payload)
-            : base(methodName)
+        /// <param name="payload">The direct method payload that will be serialized using <see cref="DefaultPayloadConvention"/>.</param>
```

Comment:
The EdgeModuleDirectMethodRequest class itself needs to be serialized using newtonsoft, so having a nested custom serializable property (Payload) within it is cumbersome to implement.
We're going to start off by restricting the usage of this to newtonsoft serializable payloads, but we can decide to expand support in the future if we like.
(This is the only type on the client side that has this kind of a "nested" serialization requirement).